src/components/editors/DictionaryLinksEditor.tsx

```diff
@@ -0,0 +1,152 @@
+import useOutsideClickListener from "../../hooks/useOutsideClickListener";
+import { useSelectedBoxStore } from "../../hooks/useSelectedBoxStore";
+import { DictionaryRelation } from "../../models/Dictionary";
+import Icon from "../Icon";
+import Select from "../util/Select";
+
+export const useLinksStyles = createUseStyles({
+		links: {
+			width: '100%',
+			fontSize: 12,
+		},
+		add: {
+			cursor: 'pointer',
+			backgroundColor: 'transparent',
+			margin: '10px 0 0',
+			border: '1px grey solid',
+			outline: 'none',
+			borderRadius: '50%',
+			padding: 0,
+			width: 24,
+			height: 24,
+			'&:hover': {
+				backgroundColor: '#e5e5e5'
+			}
+		}
+});
+
+interface DictionaryLinkProps {
+	link: DictionaryRelation;
+	deleteLink: () => void;
+}
+
+const useLinkStyle = createUseStyles({
+	link: {
+		padding: '0 2px',
+		display: 'flex',
+		justifyContent: 'space-between',
+		alignItems: 'center',
+		'&:hover': {
+			backgroundColor: '#e5e5e5'
+		}
+	},
+	title: {
+		display: 'flex',
+		'&>p': {
+			margin: '0 0 0 2px'
+		}
+	},
+	delete: {
+		display: 'inline-flex',
+		padding: '0 2px',
+		verticalAlign: 'middle',
+		backgroundColor: 'transparent',
+		outline: 'none',
+		border: 'none',
+		cursor: 'pointer',
+	}
+})
+
+const Link = ({link, deleteLink}: DictionaryLinkProps) => {
+	const classes = useLinkStyle();
+	return (
+		<div className={classes.link}>
+			<div className={classes.title}>
+				<Icon id='book' stroke='black' />
+				<p>{link.dictionary.name}</p>
+			</div>
+			<button className={classes.delete} onClick={deleteLink}><Icon id='cross' stroke='black' width={8} height={8}/></button>
+		</div>
+	)
+}
+
+const DictionaryLinksEditor = () => {
+	const classes = useLinksStyles();
+	const boxesStore = useBoxesStore();
+	const selectedBoxStore = useSelectedBoxStore();
+	const dictionaryLinksStore = useDictionaryLinksStore();
+
+	const options = useMemo(() => {
+		return boxesStore.dictionaries
+			.filter(dict => !dictionaryLinksStore.linkedDictionaries?.some(link => link.dictionary.name === dict.name))
+			.map(dict => dict.name)
+	}, [boxesStore.dictionaries, dictionaryLinksStore.linkedDictionaries])
+
+	const [showAddDictionary, setShowAddDictionary] = useState(false);
+	const [newLinkedDictionaryName, setNewLinkedDictionaryName] = useState(options[0]);
+
+	const ref = useRef<HTMLDivElement>(null);
+	useOutsideClickListener(ref, () => {
+		setShowAddDictionary(false);
+	})
+
+	
+	const applyNewLink = useCallback(() => {
+		setShowAddDictionary(false)
+		if (selectedBoxStore.box) {
+			const newLinkDictionary: DictionaryRelation = {
+				name: `${selectedBoxStore.box.name}-dictionary`,
+				box: selectedBoxStore.box.name,
+				dictionary: {
+					name: newLinkedDictionaryName,
+					type: 'MAIN'
+				}
+			}
+			dictionaryLinksStore.addLinkDictionary(newLinkDictionary);
+		}
+	}, [newLinkedDictionaryName, selectedBoxStore.box])
+
+	return dictionaryLinksStore.linkedDictionaries 
```

Comment:
same, it's type is array, it will always be truthy, either type is wrong or this check is redundant